sympy/matrices/dense.py

```diff
@@ -1386,33 +1387,36 @@ def randMatrix(r, c=None, min=0, max=99, seed=None, symmetric=False,
     >>> A == randMatrix(3, seed=1)
     True
     >>> randMatrix(3, symmetric=True, percent=50) # doctest:+SKIP
-    [0, 68, 43]
-    [0, 68,  0]
-    [0, 91, 34]
+    [77, 70,  0],
+    [70,  0,  0],
+    [ 0,  0, 88]
     """
     if c is None:
         c = r
     # Note that ``Random()`` is equivalent to ``Random(None)``
     prng = prng or random.Random(seed)
-    if symmetric and r != c:
-        raise ValueError(
-            'For symmetric matrices, r must equal c, but %i != %i' % (r, c))
+
     if not symmetric:
         m = Matrix._new(r, c, lambda i, j: prng.randint(min, max))
-    else:
-        m = zeros(r)
-        for i in range(r):
-            for j in range(i, r):
-                m[i, j] = prng.randint(min, max)
-        for i in range(r):
-            for j in range(i):
-                m[i, j] = m[j, i]
-    if percent == 100:
-        return m
-    else:
-        z = int(r*c*percent // 100)
+        if percent == 100:
+            return m
+        z = int(r * c * (100 - percent) // 100)
         m._mat[:z] = [S.Zero]*z
         prng.shuffle(m._mat)
+
+        return m
+
+    # Symmetric case
+    if r != c:
+        raise ValueError('For symmetric matrices, r must equal c, but %i != %i' % (r, c))
+    m = zeros(r)
+    ij = [(i, j) for i in range(r) for j in range(i, r)]
+    if percent != 100:
+        ij = prng.sample(ij, int(len(ij) *  percent // 100))
```

Comment:
Nice use of sample (to sample indices).

Shouldn't you use `int(len(ij)*percent/S(100))`, otherwise if percent is an int, say `3` then `3//100` will be zero and non of the matrix will be filled. It would be good to test this with a 10*10 matrix.

Reply:
No, len(ij) * percent will be executed before the division.

Where should the test go?

Reply:
```
ij = [(i,j) for i in range(10) for j in range(9)]
assert int(len(ij)*percent/S(100)) == len(ij) *  percent // 100
```

Reply:
Ahh yes. But if `len(ij)*percent` is less than 10 then you will get truncation, right?

Reply:
You'll get truncation almost always with a fix percentage, but yes, if `len(ij) * percent` is smaller than 100, you'll get the null matrix. I tried to keep the original behaviour, but while we are at it, we can round the result before casting to int. Since the current behaviour is completely broken, I think we can safely change it.

Another option would be that each entry has a `percentage` probability of being non zero, depending if you are interested in a macro or microcanonical ensemble.